[Feature]: Support Tool Calling with transformers 5.x for GLM-4.6V Models

### 🚀 The feature, motivation and pitch

Sorry but please allow me to use some AI to summary my findings and request 

**Request:** Add support for tool calling (`glm45` parser) when using transformers 5.x with GLM-4.6V models.

**Current Limitation:** GLM-4.6V models require transformers 5.0.0rc1 for vision/multimodal support, but vLLM v0.13's tool calling parser (`glm45`) is incompatible with transformers 5.x, forcing users to choose between tool calling OR vision.

## Environment

- **vLLM Version:** v0.13.0
- **Model:** `zai-org/GLM-4.6V-FP8`, `zai-org/GLM-4.6V-Flash`
- **Transformers 4.56.0:** Tool calling works, vision not supported
- **Transformers 5.0.0rc1:** Vision works, tool calling broken
- **Tool Call Parser:** `glm45`
- **GPU:** 2x NVIDIA RTX 6000 Pro (96GB each)

## Current Behavior

### With transformers 4.56.0 (Tool Calling Works)

```bash
vllm serve zai-org/GLM-4.6V-FP8 \
  --tool-call-parser glm45 \
  --limit-mm-per-prompt '{"image": 0, "video": 0}'
```

Response includes `tool_calls`:
```json
{
  "choices": [{
    "message": {
      "tool_calls": [{
        "id": "call_abc",
        "function": {
          "name": "get_weather",
          "arguments": "{\"location\": \"Tokyo\"}"
        }
      }]
    }
  }]
}
```

### With transformers 5.0.0rc1 (Tool Calling Broken)

```bash
pip install --upgrade 'transformers==5.0.0rc1'
vllm serve zai-org/GLM-4.6V-FP8 \
  --tool-call-parser glm45 \
  --limit-mm-per-prompt '{"image": 1, "video": 0}'
```

Response has `tool_calls: null`:
```json
{
  "choices": [{
    "message": {
      "content": "I should use the get_weather function...",
      "tool_calls": null
    }
  }]
}
```

## Why transformers 5.0.0rc1 is Required

GLM-4.6V models have a multimodal vision processor that only exists in transformers 5.0.0rc1:

```
Error: Cannot find module 'transformers.models.glm.visualizing_glm'
```

The `GLMVisionProcessor` class was added in transformers 5.0.0rc1 and is required for image support.

### Alternatives

_No response_

### Additional context

_No response_

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/examples/tool_chat_template_glm4.jinja
+++ b/examples/tool_chat_template_glm4.jinja
@@ -1,0 +1,54 @@
+{%- set counter = namespace(index=0) -%}
+{%- if not tools is defined %}
+    {%- set tools = none %}
+{%- endif %}
+
+{%- if messages and messages[0]['role'] == 'system' %}
+    {%- set system_message = messages[0]['content']|trim %}
+    {%- set messages = messages[1:] %}
+{%- else %}
+    {%- set system_message = "You are a helpful assistant." %}
+{%- endif %}
+
+{%- if tools is not none %}
+    {%- set tool_instruction %}
+You have access to the following tools. When you need to call a tool, you MUST use the following format:
+
+<tool_call>function_name
+<arg_key>parameter_name</arg_key>
+<arg_value>parameter_value</arg_value>
+</tool_call>
+
+Important rules:
+- Always wrap tool calls with <tool_call>...</tool_call> tags
+- Put the function name on the first line after <tool_call>
+- Use <arg_key> and <arg_value> tags for each parameter
+- If a parameter value is a string, keep it as-is. If it's a number or boolean, convert it appropriately
+- You can make multiple tool calls if needed
+- If no tool is suitable, respond with regular text
+
+Available tools:
+{% endset %}
+    {{- tool_instruction + "\n\n" }}
+    {%- for t in tools %}
+        {{- t | tojson(indent=4) }}
+        {{- "\n\n" }}
+    {%- endfor %}
+{%- endif %}
+
+{%- for message in messages -%}
+    {%- if message['role'] == 'user' -%}
+        {{- '[Round ' + counter.index|string + ']\n问：' + message['content'] -}}
+        {%- set counter.index = counter.index + 1 -%}
+    {%- endif -%}
+    {%- if message['role'] == 'assistant' -%}
+        {{- '\n答：' + message['content'] -}}
+        {%- if (loop.last and add_generation_prompt) or not loop.last -%}
+            {{- '\n' -}}
+        {%- endif -%}
+    {%- endif -%}
+{%- endfor -%}
+
+{%- if add_generation_prompt and messages[-1]['role'] != 'assistant' -%}
+    {{- '\n答：' -}}
+{%- endif -%}

--- a/vllm/tool_parsers/glm4_moe_tool_parser.py
+++ b/vllm/tool_parsers/glm4_moe_tool_parser.py
@@ -56,6 +56,20 @@ class Glm4MoeModelToolParser(ToolParser):
         self.tool_call_end_token_id = self.vocab.get(self.tool_call_end_token)
         self._buffer = ""
 
+    def adjust_request(self, request: ChatCompletionRequest) -> ChatCompletionRequest:
+        """
+        Adjust request parameters to ensure tool call tokens are not skipped
+        during tokenizer decoding.
+        """
+        request = super().adjust_request(request)
+        if request.tools and request.tool_choice != "none":
+            # Ensure tool call tokens (<tool_call>, </tool_call>) are not skipped
+            # during decoding. Even though they are not marked as special tokens,
+            # setting skip_special_tokens=False ensures proper handling in
+            # transformers 5.x where decoding behavior may have changed.
+            request.skip_special_tokens = False
+        return request
+
     def extract_tool_calls(
         self,
         model_output: str,

